### DISCUSSION TITLE
Enable cycling through completions

### DISCUSSION THREAD

**Discussion by volodymyrprokopyuk** [company-mode/company-mode#1421] [Q&A] [answered]
Hi,

I'm using the latest doom-emacs with Emacs 29.

I want to enable cycling through completions: being on the first candidate and invoking the `company-select-previous` goes to the last candidate in the completion list and being on the last candidate and invoking the `company-select-next` goes to the first candidate in the completion list. How this can be achieved?

My current configuration of the `company-mode` is

```elisp
  (add-hook 'after-init-hook #'global-company-mode)
  (evil-define-key nil company-active-map (kbd "M-m") #'company-select-next)
  (evil-define-key nil company-active-map (kbd "M-,") #'company-select-previous)
```

Thank you!

**Comment by dgutov** [comment, marked as answer]
Hi!

The user option to customize is `company-selection-wrap-around`.

To find all the available options, type <kbd>M-x customize-group RET company RET</kbd>.

**Reply by volodymyrprokopyuk** [reply]
Thank you very much! `(setq company-selection-wrap-around 1)` worked for me!
